### ISSUE TITLE
Class not found issue `com.github.dockerjava.core.DefaultDockerClientConfig`

### ISSUE THREAD

**Issue by liviutudor** [nebula-plugins/nebula-docker-plugin#4]
When using `dockerImage` closure, the docker plugin has an issue finding class `com.github.dockerjava.core.DefaultDockerClientConfig`:

```
[23:01:44] [metrics] Build finished with result Result(status=FAILURE, failures=[org.gradle.internal.exceptions.LocationAwareException: Execution failed for task ':buildImage'.
[23:01:44] 	at org.gradle.initialization.DefaultExceptionAnalyser.transform(DefaultExceptionAnalyser.java:74)
[23:01:44] 	at org.gradle.initialization.MultipleBuildFailuresExceptionAnalyser.transform(MultipleBuildFailuresExceptionAnalyser.java:47)
[23:01:44] 	at org.gradle.initialization.StackTraceSanitizingExceptionAnalyser.transform(StackTraceSanitizingExceptionAnalyser.java:30)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:101)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
[23:01:44] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
[23:01:44] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
[23:01:44] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
[23:01:44] 	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
[23:01:44] 	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
[23:01:44] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
[23:01:44] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
[23:01:44] 	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:81)
[23:01:44] 	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:46)
[23:01:44] 	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:51)
[23:01:44] 	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:28)
[23:01:44] 	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:43)
[23:01:44] 	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:173)
[23:01:44] 	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:241)
[23:01:44] 	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:214)
[23:01:44] 	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
[23:01:44] 	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
[23:01:44] 	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:207)
[23:01:44] 	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
[23:01:44] 	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
[23:01:44] 	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
[23:01:44] 	at org.gradle.launcher.Main.doAction(Main.java:33)
[23:01:44] 	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
[23:01:44] 	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:55)
[23:01:44] 	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:36)
[23:01:44] 	at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
[23:01:44] 	at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
[23:01:44] 	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
[23:01:44] 	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
[23:01:44] Caused by: org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':buildImage'.
[23:01:44] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:68)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
[23:01:44] 	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
[23:01:44] 	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
[23:01:44] 	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
[23:01:44] 	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
[23:01:44] 	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
[23:01:44] 	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
[23:01:44] 	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
[23:01:44] 	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
[23:01:44] 	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
[23:01:44] 	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
[23:01:44] 	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
[23:01:44] 	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
[23:01:44] 	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
[23:01:44] 	at org.gradle.internal.Factories$1.create(Factories.java:22)
[23:01:44] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
[23:01:44] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
[23:01:44] 	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
[23:01:44] 	... 32 more
[23:01:44] Caused by: org.gradle.internal.UncheckedException: java.lang.ClassNotFoundException: com.github.dockerjava.core.DefaultDockerClientConfig
[23:01:44] 	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:45)
[23:01:44] 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:78)
[23:01:44] 	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:228)
[23:01:44] 	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:221)
[23:01:44] 	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:210)
[23:01:44] 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
[23:01:44] 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
[23:01:44] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
[23:01:44] 	... 60 more
[23:01:44] Caused by: java.lang.ClassNotFoundException: com.github.dockerjava.core.DefaultDockerClientConfig
[23:01:44] 	at java_lang_ClassLoader$loadClass.call(Unknown Source)
[23:01:44] 	at com.bmuschko.gradle.docker.utils.DockerThreadContextClassLoader.loadClass(DockerThreadContextClassLoader.groovy:133)
[23:01:44] 	at com.bmuschko.gradle.docker.utils.DockerThreadContextClassLoader.getDockerClient(DockerThreadContextClassLoader.groovy:103)
[23:01:44] 	at com.bmuschko.gradle.docker.utils.DockerThreadContextClassLoader.withClasspath(DockerThreadContextClassLoader.groovy:54)
[23:01:44] 	at com.bmuschko.gradle.docker.utils.ThreadContextClassLoader$withClasspath.call(Unknown Source)
[23:01:44] 	at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask.runInDockerClassPath(AbstractDockerRemoteApiTask.groovy:62)
[23:01:44] 	at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask$runInDockerClassPath.callCurrent(Unknown Source)
[23:01:44] 	at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask.start(AbstractDockerRemoteApiTask.groovy:56)
[23:01:44] 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
[23:01:44] 	... 67 more
[23:01:44] ])
```

Adding explicitely the `docker-java` library to classpath doesn't help:

```groovy
buildscript {
    repositories {
        jcenter()
    }

    dependencies {
        classpath 'com.github.docker-java:docker-java:3.0.3'
        classpath 'com.bmuschko:gradle-docker-plugin:3.0.3'
        classpath "com.netflix.nebula:nebula-docker-plugin:$nebulaDockerPluginVersion"
    }
}
```

**Comment by liviutudor** [comment]
Culprit is in this class : https://github.com/bmuschko/gradle-docker-plugin/blob/master/src/main/groovy/com/bmuschko/gradle/docker/utils/DockerThreadContextClassLoader.groovy

which references this library: https://github.com/docker-java/docker-java

**Comment by liviutudor** [comment]
found solution as per: http://liviutudor.com/2016/11/15/gotcha-when-building-gradle-plugins-on-top-of-docker-java-plugin/#sthash.Sb2KNHzs.dpbs

**Comment by liviutudor** [comment]
this hasbeen fixed in version `1.1.0`
